x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/InTests.java

```diff
@@ -0,0 +1,289 @@
+            }
+            args.add(new TestCaseSupplier.TypedData(field, EsqlDataTypes.GEO_SHAPE, "field"));
+            return new TestCaseSupplier.TestCase(
+                args,
+                matchesPattern("InExpressionEvaluator\\[EqualsGeometriesEvaluator.*\\]"),
+                DataTypes.BOOLEAN,
+                equalTo(inlist.contains(field))
+            );
+        }));
+
+        suppliers.add(new TestCaseSupplier("cartesian_point", List.of(EsqlDataTypes.CARTESIAN_POINT, EsqlDataTypes.CARTESIAN_POINT), () -> {
+            List<Object> inlist = randomList(items, items, () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomPoint())));
+            Object field = new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomPoint()));
+            List<TestCaseSupplier.TypedData> args = new ArrayList<>(inlist.size() + 1);
+            for (Object i : inlist) {
+                args.add(new TestCaseSupplier.TypedData(i, EsqlDataTypes.CARTESIAN_POINT, "inlist" + i));
+            }
+            args.add(new TestCaseSupplier.TypedData(field, EsqlDataTypes.CARTESIAN_POINT, "field"));
+            return new TestCaseSupplier.TestCase(
+                args,
+                matchesPattern("InExpressionEvaluator\\[EqualsGeometriesEvaluator.*\\]"),
+                DataTypes.BOOLEAN,
+                equalTo(inlist.contains(field))
+            );
+        }));
+
+        suppliers.add(new TestCaseSupplier("cartesian_shape", List.of(EsqlDataTypes.CARTESIAN_SHAPE, EsqlDataTypes.CARTESIAN_SHAPE), () -> {
+            List<Object> inlist = randomList(
+                items,
+                items,
+                () -> new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomGeometry(randomBoolean())))
+            );
+            Object field = new BytesRef(CARTESIAN.asWkt(ShapeTestUtils.randomGeometry(randomBoolean())));
+            List<TestCaseSupplier.TypedData> args = new ArrayList<>(inlist.size() + 1);
+            for (Object i : inlist) {
+                args.add(new TestCaseSupplier.TypedData(i, EsqlDataTypes.CARTESIAN_SHAPE, "inlist" + i));
+            }
+            args.add(new TestCaseSupplier.TypedData(field, EsqlDataTypes.CARTESIAN_SHAPE, "field"));
+            return new TestCaseSupplier.TestCase(
+                args,
+                matchesPattern("InExpressionEvaluator\\[EqualsGeometriesEvaluator.*\\]"),
+                DataTypes.BOOLEAN,
+                equalTo(inlist.contains(field))
+            );
+        }));
+    }
+
+    @Override
+    protected Expression build(Source source, List<Expression> args) {
+        return new In(source, args.get(args.size() - 1), args.subList(0, args.size() - 1));
+    }
+
+    @Override
+    public void testSimpleWithNulls() {
+        assumeFalse("test case is invalid", false);
+    }
+
+    @Override
+    public void testFactoryToString() {
+        assumeFalse("test case is invalid", false);
+    }
```

Comment:
Discussed with Nik about this, skipping testFactoryToString is for InTest only, it is because the way In is implemented in ES|QL is different from the other functions, it doesn't implement EvaluatorMapper. A follow up PR will be opened to rewrite In using EvaluatorMapper. This PR is mainly for doc generation through annotations, leave this test skipped for now, and when the In is rewritten with EvaluatorMapper, InTests will be modified.